### ISSUE TITLE
Shipping fees not updating when I create an order from the backend

### ISSUE THREAD

**Issue by sstead** [openfoodfoundation/openfoodnetwork#2513]
## Description
When you create an order in the backend there's a screen where you select the shipping method. When you first get to the page the first shipping method is selected by default and the charge associated with that shipping method shows. If I change to another shipping method with another fee the charge doesn't update. 

If I select to use another shipping method, with the fee not updated, when I go to apply a payment method to the order, the total is not correct becuase it's now showing the right shipping method fee. After I apply the payment the correct Total updates, including the right fee. The implication is that I might process one payment amount (using Pin/Stripe where the amount is charged), only to then have the total update. This can leave me with more balance due (so I need to process a second payment - which could have fee implications with Stripe/Pin), or I have charged too much and there's credit owing.

## Expected Behavior
I would expect the total to update correctly when I change between shipping methods with different fees.

## Actual Behavior
It's easy enough to end up with the wrong shipping method fee applied.

## Steps to Reproduce
1. Make sure a shop has 2 shipping methods with different fees
2. Go to the backend and create an order for a customer in that shop. Look at the base order price
3. Go to the shipping method section, and change between the shipping methods. Note that the charge won't update. Select a shipping method where the charge hasn't updated correctly.
4. In the payments area apply the payment using any payment method. 
5. See the confirmation page where there's either balance due or credit owed, because only now has the system seen the correct shipping fee.

## Animated Gif/Screenshot
![image](https://user-images.githubusercontent.com/10843056/43617380-13823310-9705-11e8-997b-06e97c63c755.png)

https://www.useloom.com/share/71517fcfce3b4a70ae61846f2ffaabfe  

## Context
Found while testing https://github.com/openfoodfoundation/openfoodnetwork/pull/2420
This bug can occur quite easily in a shop with multiple shipping methods with different fees.

## Severity
s3

## Your Environment
Chrome, PC, staging1 and openfoodnetwork.org.au

## Possible Fix
Make the order total update at the time the shipping method is selected, so that the payment total is correct.

**Comment by filipefurtad0** [comment]
Thanks for raising this @sigmundpetersen and @achauve . Indeed, there have been quite some changes ever since this issue was opened.

Currently, this is possible to update the shipping fee, but it is necessary to select `Yes` on "Associated adjustment closed":

![Peek 2021-10-18 11-44](https://user-images.githubusercontent.com/49817236/137716742-d17dae6a-82ce-4d58-bea4-e3df083b183f.gif)

The above example is an order in which we have:
- bike - fee $5
- pick up - fee $1
- Tax: 13%

Only after doing so is the value of the new shipping method considered.

It's important to note that only after clicking "Update and Recalculate Fees" we get the correct tax calculation on the new fee.

This is a bit hacky, and perhaps the UI could be improved, as it can easily lead to wrong values. For example, if one:
- changes the shipping method
- keeps `No` on "Associated adjustment closed"
- Clicks "Update and Recalculate Fees"

In this case, the new shipping fee will be considered for tax calculation, but not the new shipping fee.

So, it's only if we keep `Yes` on "Associated adjustment closed", that it all works as expected. I'm not really sure why we need to keep this button. Should we just set `Yes` as default and remove "Associated adjustment closed" (on a separate issue)?

I've met this before - not sure on the outcome on the discussion. What do you think @RachL ?

**Comment by filipefurtad0** [comment]
Thanks for pointing that out. So I guess we're observing the expected behavior:
> NO: this will keep the cost of shipping the same as the method chosen by the customer at checkout

Agree, the inconsistency is probably related to translations.

The use-case is then, the user wishes to change the shipping method of the order, but keep the shipping fee. This was confusing to me, because the amount of the new fee is still displayed, right next to the previous one - but the user guide clarifies it :+1: 

The discrepancy between taxes is only related to the need to click "Update and Recalculate Fees", which is not a bug I guess.

So, I guess we can close this issue?
